### ISSUE TITLE
`--disallow-untyped-calls` reports errors in untyped functions

### ISSUE THREAD

**Issue by pranavrajpal** [python/mypy#12912]
**Bug Report**

Using `--disallow-untyped-calls` and `--check-untyped-defs` results in errors about calling untyped functions in a typed context, even when calling from an untyped function.

**To Reproduce**

Test case:
```py
[case testDisallowUntypedCallsFromUntypedFunction]
# flags: --disallow-untyped-calls --check-untyped-defs

def f(x):
    pass

def g(y):
    f(y)
```
**Expected Behavior**

Either:
- the documentation and the error message should be updated to specify that `--disallow-untyped-calls` disallows all calls to untyped functions, regardless of whether the current function has type annotations, or
- the above code shouldn't produce any errors

**Actual Behavior**

The call to `f` in the above example produces the error "Call to untyped function "f" in typed context".

**Your Environment**

See https://mypy-play.net/?mypy=latest&python=3.10&flags=check-untyped-defs%2Cdisallow-untyped-calls&gist=3350fe27968b29dabc5ef1e207d008ce to reproduce.

This came up in #12901 (specifically the mypy_primer diff had an error [here](https://github.com/aiortc/aiortc/blob/25de502661f0603bee256e91e6e47633b1336ea7/src/aiortc/contrib/media.py#L74))
